darwin/future/core/types/common.py

```diff
@@ -1,13 +1,20 @@
 from __future__ import annotations
 
-from typing import Any, Dict, List, Union
+from typing import Any, Dict, List, Mapping, Protocol, Union
 
 from darwin.future.data_objects import validators as darwin_validators
-from darwin.future.data_objects.typing import UnknownType
 
 JSONType = Union[Dict[str, Any], List[Dict[str, Any]]]  # type: ignore
 
 
+class Implements_str(Protocol):
```

Comment:
so this protocol should mean it'll work with anything that can be converted to a string, so things like ints, floats etc will work.